cmsl1t/analyzers/BaseAnalyzer.py

```diff
@@ -11,18 +11,20 @@ class BaseAnalyzer(object):
     A Base class to be used by the various analyzers
     """
 
-    def __init__(self, name, config, **kwargs):
-        self.name = name
-        self.output_folder = config.get('output', 'folder')
-        self.plots_folder = config.get('output', 'plots_folder')
-        self.config = config
+    def __init__(self, **kwargs):
+        self.name = kwargs.pop('name')
+        self.output_folder = kwargs.pop('output_folder')
+        self.plots_folder = kwargs.pop('plots_folder')
+        self.all_plots = []
+        self.puBins = kwargs.pop('puBins')
+        self.thresholds = kwargs.pop('thresholds')
+        self.file_format = kwargs.pop('file_format')
+        self.params = kwargs
```

Comment:
`self.params` is supposed to be accessed directly, but that makes it relatively to easy to manipulate.  Would it be safer if it were made a read-only `@property`?

Reply:
Yes, that's a good idea

Reply:
Done:
```python
params = property(fget=lambda x: x.__params, fset=None)
```